Dockerfile

```diff
@@ -0,0 +1,78 @@
+
+FROM ubuntu:16.04
+
+ARG BUILDBOT
+ARG PUB_KEY
+ARG PRIV_KEY
+ARG CONFIG
+ARG BUILBOT_MAIL
+ARG PROD
+ARG MACHINE
+ARG BRANCH
+
+RUN apt-get update
+# install the requiured prerequisites
+RUN apt-get install gcc-5-aarch64-linux-gnu gcc-aarch64-linux-gnu -y
+RUN apt-get remove gcc-5-multilib gcc-multilib
+RUN apt-get install -y vim \
+    cpio gawk wget diffstat texinfo chrpath socat libsdl1.2-dev \
+    python-crypto repo checkpolicy python-git python-github \
+    python-ctypeslib bzr pigz m4 lftp openjdk-8-jdk git-core \
+    gnupg flex bison gperf build-essential zip curl zlib1g-dev \
+    gcc-multilib g++-multilib libc6-dev-i386 lib32ncurses5-dev \
+    x11proto-core-dev libx11-dev lib32z-dev ccache libgl1-mesa-dev \
+    libxml2-utils xsltproc unzip python-clang-5.0 gcc-5 g++-5 bc \
+    python-pip python3-pyelftools python3-pip python3-crypto \
+    android-tools-fsutils libssl-dev \
+    sudo policykit-1 locales && \
+    dpkg-reconfigure locales  && \
+    locale-gen en_US.UTF-8 && \
+    update-locale LC_ALL=en_US.UTF-8 LANG=en_US.UTF-8 && \
+    pip install pycryptodome && \
+    pip3 install pycryptodomex && \
+    mkdir -p /mnt/work/build && \
+    if [ "$BUILDBOT" != "root" ]; then \
+    useradd -ms /bin/bash "$BUILDBOT" && \
+    passwd -d $BUILDBOT && \
+    usermod -aG sudo "$BUILDBOT" && \
+    mkdir -p /home/"$BUILDBOT"/.ssh && \
+    chown -R "$BUILDBOT" /home/"$BUILDBOT" ; \
+    fi
+
+RUN chown -R "$BUILDBOT" /mnt/work && \
+    chown -R "$BUILDBOT" /usr/bin && \
+    chown root:root /usr/bin/sudo && chmod 4755 /usr/bin/sudo && \
+    echo "$BUILDBOT ALL=(ALL) NOPASSWD:ALL" >> /etc/sudoers
+
+COPY "$PRIV_KEY" /root/.ssh/
+COPY "$PUB_KEY" /root/.ssh/
+
+USER "$BUILDBOT"
+
+RUN curl https://storage.googleapis.com/git-repo-downloads/repo-1 > ~/repo && \
+    chmod a+x ~/repo && \
+    sudo cp ~/repo /usr/bin/ 
```

Comment:
Do we really need this sudo? (It used only once here). Do we need to have repo in /usr/bin ? Could we place it in some /tmp/bin/repo and use `export PATH=/tmp/bin:$PATH` instead ?

Reply:
regarding the PATH, in this case, it does not work, because the PATH processing comes from the begin-to-end, so, the 1st repo (old one) will start - this is not expected result. I believe sudo can be omitted you are right.